### ISSUE TITLE
Unable to propagate state changes to a child "SelectComponent"

### ISSUE THREAD

**Issue by alvsan09** [eclipse-theia/theia#11050]
### Bug Description:
While trying to re-base the open PR which introduces dynamic debug configurations https://github.com/eclipse-theia/theia/pull/10212
to the latest master, the solution needs to integrate with the recent introduction of 'Custom select component' https://github.com/eclipse-theia/theia/pull/10991, this can be achieved by delegating the rendering of the drop down to the new component while keeping the dynamics of refreshing configurations on a parent component.

Using the new "SelectComponent" as a child requires to be able to propagate state changes from the parent component, this is currently failing.

### Steps to Reproduce:
Using branch: `asl/dynamic-debug-config-view-draft`
commit: e1020bbc261

1. Open the debug view
2. Select the dynamic provider type `pwa-node` and then from the quick pick select `Create Java Script Debug Terminal`
3. Observe that the selected item in the text box is `pwa-node` rather than the selected one.

![selectComponentNotUpdated](https://user-images.githubusercontent.com/76971376/164374814-2991cc11-304f-49d8-8abd-4d6eab61f6c8.gif)

### Additional Information
The propagation of the state is expected as the state of the parent component updates the current configuration with the selected one,
and therefore triggers a call to the render function, however the render function does not create a new instance of the 'SelectComponent` and therefore the state on the child component remains unchanged.

An attempt has been made to derive the state from the updated properties by using the static method for react components `getDerivedStateFromProps` as per: https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

However this has side effects e.g. The hovering function does not longer work
This attempt can be observed using the same branch mentioned above:
commit: 8e40c864063

- Operating System: Ubuntu
- Theia Version: commit: 1972f21c93e

**Comment by msujew** [comment]
@alvsan09 While I'm sure there's a "correct" react way to approach this issue, it seems like a small workaround is able to fix this issue. See my commit https://github.com/eclipse-theia/theia/commit/da6d15b7f7052f61c1f317979dd87160c0a9bf84.

**Comment by alvsan09** [comment]
@msujew, your opinion on the above will be really valuable to move forward with the support of dynamic debug configurations.

**Comment by msujew** [comment]
@alvsan09 I agree, doing this without stopping the data flow is definitely preferable. I'm happy for any changes to the `SelectComponent`, but currently don't have the time to implement them myself.
